### DISCUSSION TITLE
vi Directory Questions

### DISCUSSION THREAD

**Discussion by joegittingherdone** [open-watcom/open-watcom-v2#683] [Q&A]
I was wondering whether someone could help me understand the `bld/vi` directory structure.  I want to explore/modify code for the viw.exe.  I am using 64-bit Windows 7.  What directories should I be looking at?  And what about for 32-bit Windows XP?  I have been using wmake in some, but I am not sure I'm looking at this correctly because I don't see my code modifications show up in the executable.

And while we're on the topic, it seems like the `rel `directory is similar to `bld/vi`; so what's the difference?

**Comment by jmalak** [comment]
On the Wiki is some basic information about [OW2 build](https://github.com/open-watcom/open-watcom-v2/wiki/Build) directory tree structure and also [developer documentation](https://open-watcom.github.io/open-watcom-v2-wikidocs/devguide.html).
`rel` directory can be change to any outside OW2 directory tree and it contains full OW2 build in structure how it is installed on host system.
`bld/vi` contains all sources related to vi project and also temporary files for build process.

**Comment by joegittingherdone** [comment]
Thank you.  I forgot how confusing the naming of Watcom's directories was.  I looked at your links and they didn't answer my questions, so I explored the directories and have guesses as to what they are:

In `bld`:
`dos386` = 32-bit DOS/character mode
`nt386` = 32-bit DOS/character mode
`nt386.win` = 32-bit Windows
`ntx64` = 64-bit DOS/character mode*
`ntx64.win` = 64-bit Windows*
*I assume the x64 directories require the VC++ compiler.

In `rel`:
`binnt` = 32-bit Windows
`binnt64` = 64-bit Windows
`binw` = 16-bit Windows

**Reply by jmalak** [reply]
directories name are for build objects has name with following general syntax 'target OS' + 'target CPU'
sometimes suffix is used as in vi '.win' is GUI version of vi otherwise it is character mode vi.

If you want 64-bit Windows version then you need appropriate version of VC++ compiler.
rel is standard installation how it is install on host system by OW installer and each OS + architecture has separated directory for executable.
binw = common objects, 16/32-bit DOS, 16-bit Windows etc.
binnt = 32-bit Windows
binnt64 = 64-bit Windows
binp = 16/32-bit OS/2
binl = 32-bit Linux
binl64 = 64-bit linux
....

**Comment by joegittingherdone** [comment]
Thank you.
